plugin/src/main/java/io/jenkins/plugins/analysis/core/model/RunTab.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDisplayName() {
+        return "Warnings";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getUrlName() {
+        return "warnings";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Badge getBadge() {
+        var warningActionsCount = getWarningActions().stream().map(e -> e.getResult().getTotalSize()).reduce(0, Integer::sum);
+
+        if (warningActionsCount == 0) {
+            return null;
+        }
+
+        return new Badge(String.valueOf(warningActionsCount), Messages.ResultAction_Badge(warningActionsCount), Badge.Severity.WARNING);
+    }
+
+    /**
+     * The list of warning actions belonging to a run.
+     * @return the list of ordered warning actions, sorted by warning count, then alphabetized.
+     */
+    @Restricted(NoExternalUse.class)
+    public List<ResultAction> getWarningActions() {
+        return getObject()
+                .getActions(ResultAction.class)
+                .stream()
+                .sorted(Comparator
+                        .comparingInt((ResultAction a) -> a.getResult().getTotalSize())
+                        .reversed()
+                        .thenComparing(ResultAction::getDisplayName, String.CASE_INSENSITIVE_ORDER)
+                )
+                .toList();
+    }
+
+    /**
+     * Renders the default view of the Warnings tab.
+     *
+     * @param req the incoming Stapler request
+     * @param rsp the HTTP response
+     * @throws IOException if an I/O error occurs during forwarding
+     * @throws ServletException if forwarding fails
+     */
+    public void doIndex(final StaplerRequest2 req, final StaplerResponse2 rsp) throws IOException, ServletException {
+        req.setAttribute("resultAction", getWarningActions().get(0));
+        req.getView(this, "index.jelly").forward(req, rsp);
+    }
+
+    /**
+     * Renders a dynamic subpage of the Warnings tab.
+     *
+     * <p>
+     * This method is bound to the remainder of the URL path after the tab's root.
+     * It looks up the {@link ResultAction} that matches the requested URL segment
+     * and, if found, forwards the request to {@code index.jelly} with that action
+     * as the backing model.
+     *
+     * @param req the incoming Stapler request
+     * @param rsp the HTTP response
+     * @throws IOException if an I/O error occurs during forwarding
+     * @throws ServletException if forwarding fails
+     */
+    public void doDynamic(final StaplerRequest2 req, final StaplerResponse2 rsp) throws IOException, ServletException {
+        String action = req.getRestOfPath().substring(1);
+        ResultAction resultAction = null;
+
+        for (ResultAction ui : getWarningActions()) {
+            String urlName = ui.getUrlName();
+            if (urlName != null && urlName.equals(action)) {
+                resultAction = ui;
```

Comment:
Rather than storing the action it would make sense to delegate to the new view here:
```
        req.setAttribute("resultAction", resultAction);
        req.getView(this, "index.jelly").forward(req, rsp);
```
Then we don't need the if block later